frontend/src/main/scala/bloop/engine/BuildLoader.scala

```diff
@@ -80,16 +165,60 @@ object BuildLoader {
   def loadSynchronously(
       configDir: AbsolutePath,
       logger: Logger
-  ): List[Project] = {
+  ): List[LoadedProject] = {
+    val settings = WorkspaceSettings.readFromFile(configDir, logger)
     val configFiles = readConfigurationFilesInBase(configDir, logger).map { ap =>
       val bytes = ap.path.readAllBytes
       val hash = ByteHasher.hashBytes(bytes)
       Build.ReadConfiguration(Origin(ap, hash), bytes)
     }
 
     logger.debug(s"Loading ${configFiles.length} projects from '${configDir.syntax}'...")(
-      DebugFilter.Compilation
+      DebugFilter.All
     )
-    configFiles.map(f => Project.fromBytesAndOrigin(f.bytes, f.origin, logger))
+
+    configFiles.map { f =>
```

Comment:
I don't like this deduplicated logic but I don't mind living with it. There's no way to reuse the logic in `loadBuildIncrementally` because everything there runs in parallel and it's implemented in terms of `Task`.